process/handlers.lua

```diff
@@ -116,28 +164,80 @@ function handlers.after(handleName)
 end
 
 function handlers.remove(name)
-  assert(type(name) == 'string', 'name MUST be string')
-  if #handlers.list == 1 and handlers.list[1].name == name then
-    handlers.list = {}
-    
+  assert(type(name) == 'string', 'Handler name MUST be a string')
+  
+  local idx = findIndexByProp(handlers.list, "name", name)
+  if idx then
+    table.remove(handlers.list, idx)
+  else
+    error('Handler not found')
   end
+end
 
-  local idx = findIndexByProp(handlers.list, "name", name)
-  table.remove(handlers.list, idx)
-  
+function handlers.matchesPattern(msg, pattern)
+  if type(pattern) == 'function' then
+    return pattern(msg)
+  -- If the pattern is a table, step through every key/value pair in the pattern and check if the msg matches
+  -- Supported match types:
+  --   - Exact string match
+  --   - Lua gmatch string
+  --   - '_' (wildcard: Message has tag, but can be any value)
+  --   - Function execution on the tag, optionally using the msg as the second argument
+  end
+  if type(pattern) == 'table' then
+    for key, patternMatchSpec in pairs(pattern) do
+      local matched = false
+      -- If the key is not in the message, then it does not match
+      if(not msg[key]) then
+        return false
+      end
+      -- if the patternMatchSpec is a wildcard, then it always matches
+      if patternMatchSpec == '_' then
+        matched = true
+      end
+      -- if the patternMatchSpec is a function, then it is executed on the tag value
+      if type(patternMatchSpec) == "function" then
+        if patternMatchSpec(msg[key], msg) then
+          matched = true
+        else
+          return false
+        end
+      end
+      -- if the patternMatchSpec is a string, check it for special symbols (less `-` alone)
+      -- and exact string match mode
+      if not matched and string.match(patternMatchSpec, "[%^%$%(%)%%%.%[%]%*%+%?]") then
+        if string.match(msg[key], patternMatchSpec) then
+          matched = true
+        end
+      else
+        if msg[key] == patternMatchSpec then
+          matched = true
+        end
+      end
+      -- if the patternMatchSpec is not matched, then the msg does not match
+      if not matched then
+        return false
```

Comment:
Same behavior in short-circuiting checking other `MatchSpec`s after the first non-matched spec.

Reply:
Great catch!!

Yes, this logic should actually be inverted:
- Every tag in the _matchspec_ must match. If any of them don't, we can short curcuit and return false early.
- Tags in the message that are not matched explicitly are assumed to. For example, `match(spec = {}, msg { [lots of tags]}) => true.`

Thank you for spotting this!